### PR TITLE
make replace layer code vector mask aware

### PR DESCRIPTION
**Pull request by pineapplespatula**
fix for issue #3436

### CODE REVIEW

File: src/js/models/layerstructure.js [hunk 1]
```diff
@@ -1035,7 +1035,7 @@ define(function (require, exports, module) {
             replace = replaceLayer &&
```
Comment (volfied) [inline src/js/models/layerstructure.js]: This expression hurt my brain, can you clean it up a little?

I advise: `replace = replace || (replaceLayer && !replaceLayer.... && ...)` so we can short circuit if replace is already provided.

Also, please update the comment above so it's not stale.

**Comment by volfied** [comment]
One nit, @pineapplespatula

File: src/js/models/layerstructure.js (hunk 1 above)
Comment (pineapplespatula) [inline src/js/models/layerstructure.js]: that changes the logic, replace should be false if there is no replaceLayer.

**Comment by pineapplespatula** [comment]
back to @volfied i broke out the default replacement logic into a separate variable to make it more readable

File: src/js/models/layerstructure.js [hunk 2]
```diff
@@ -1032,10 +1032,9 @@ define(function (require, exports, module) {
             }
 
             // The selected layer should be empty and a non-background layer unless replace is explicitly provided true
-            replace = replaceLayer &&
-                (replace ||
-                (!replaceLayer.isBackground && replaceLayer.isPixel &&
-                replaceLayer.bounds && !replaceLayer.bounds.area));
+            var layerShouldBeReplaced = (!replaceLayer.isBackground && replaceLayer.isPixel &&
+                    replaceLayer.bounds && !replaceLayer.bounds.area && !replaceLayer.vectorMaskEnabled);
+            replace = replaceLayer && (replace || layerShouldBeReplaced);
```
Comment (volfied) [inline src/js/models/layerstructure.js]: I think the `replaceLayer` needs to be part of layerShouldBeReplaced, if `replaceLayer` is null, above code will break.

**Comment by volfied** [comment]
:cyclone: @pineapplespatula

**Comment by pineapplespatula** [comment]
:repeat: @volfied

**Pull request merged**